x-pack/platform/plugins/shared/streams/server/lib/streams/state_management/streams/wired_stream.ts

```diff
@@ -651,6 +648,18 @@ export class WiredStream extends StreamActiveRecord<Streams.WiredStream.Definiti
     return this._definition.ingest.failure_store;
   }
 
+  private getInheritedFailureStoreFromAncestors(desiredState: State): FailureStore {
+    const ancestorsAndSelf = getAncestorsAndSelf(this._definition.name).reverse();
+    for (const ancestor of ancestorsAndSelf) {
+      const ancestorStream = desiredState.get(ancestor)! as WiredStream;
+      const ancestorFailureStore = ancestorStream.getFailureStore();
+      if (!isInheritFailureStore(ancestorFailureStore)) {
+        return ancestorFailureStore;
+      }
+    }
+    return { inherit: {} };
```

Comment:
I think this is the error condition we're looking for, a resolved wired stream lifecycle should never return `inherit`. We should throw here